### PR TITLE
feat: workspace health default-value fix

### LINKED ISSUE #3842: feat: add ability to fix default-value conflicts to `workspace:health-check` command
## Scope & Context

Main ticket is #3773 
The goal of this sub-ticket is to fix default-value conflicts, when this kind of error happen we should create the proper migration to fix them.
Some type can have complex default-value like `enum`; we must also ensure in health check that the default value is in the options of the enum.

## Expected behavior

`--fix=default-value ` can be called and fix the default-value conflict.

### PR DESCRIPTION
**Pull request by magrinj**
This PR is implementing the `default-value` kind in the --fix option of workspace:health command.
The core was created here #3863 

Fix #3842

### CODE REVIEW

File: packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  async fix(
+    manager: EntityManager,
+    objectMetadataCollection: ObjectMetadataEntity[],
+    issues: WorkspaceHealthDefaultValueIssue[],
+  ): Promise<Partial<WorkspaceMigrationEntity>[]> {
+    const workspaceMigrations: Partial<WorkspaceMigrationEntity>[] = [];
+
+    for (const issue of issues) {
+      switch (issue.type) {
+        case WorkspaceHealthIssueType.COLUMN_DEFAULT_VALUE_CONFLICT: {
```
Comment (charlesBochet) [inline packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts:34]: why introducing a switch to fix one type of issue?

File: packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  ): FieldMetadataDefaultValue<'default'> {
+    if (
+      columnDefault === undefined ||
+      columnDefault === null ||
+      columnDefault === 'NULL'
+    ) {
+      return null;
+    }
+
+    if (!isNaN(Number(columnDefault))) {
```
Comment (charlesBochet) [inline packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts:90]: is this logic duplicated with the sync script?

Comment (magrinj) [reply packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts:90]: No as we're doing the opposite here, creating a `defaultValue` from the `columnDefault` in the database

File: packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts (hunk 1 above)
Comment (magrinj) [reply packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts:34]: I was planning at first fixing multiple similar issues in one service, but didn't got the case for now

File: packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    });
+
+    return this.workspaceMigrationFieldFactory.create(
+      objectMetadataCollection,
+      fieldMetadataUpdateCollection,
+      WorkspaceMigrationBuilderAction.UPDATE,
+    );
+  }
+
+  private createDefaultValueFromColumnDefault(
```
Comment (charlesBochet) [inline packages/twenty-server/src/workspace/workspace-health/services/workspace-fix-default-value.service.ts:79]: computeFieldMetadataDefaultValueFromColumnDefault

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**